### PR TITLE
Implementation of unit tests for the second half

### LINKED ISSUE #5: Add unit tests for the second half of chosen files
Add unit tests for:

- [ ] `toNumber.js`
- [ ] `toString.js`
- [ ] `isEmpty.js`
- [ ] `isArrayLike.js`
- [ ] `drop.js`

Unit tests should include negative and edge tests as well!

### PR DESCRIPTION
**Pull request by joniriikola**
Closes #5

### CODE REVIEW

File: src/__tests__/drop.test.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import drop from '../drop';
+
+describe('drop', () => {
+    describe('positive tests', () => {
+        describe('common array cases', () => {
+            test('should drop the default 1 element from the beginning', () => {
```
Comment (erictelkkala) [inline src/__tests__/drop.test.js:6]:
```suggestion
            test('should drop the first element from the beginning by default, if no parameter is provided', () => {
```

File: src/__tests__/drop.test.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    
+        test('should return an empty array if array is not an array (e.g., number)', () => {
+            expect(drop(123, 1)).toEqual([])
+        })
+    
+        test('should return an empty array if array is not an array (e.g., object)', () => {
+            expect(drop({ a: 1, b: 2 }, 1)).toEqual([])
+        })
+    
+        test('should return an empty array if n is NaN', () => {
```
Comment (erictelkkala) [inline src/__tests__/drop.test.js:53]:
```suggestion
        test('should return the original array if n is NaN', () => {
```
Is this the correct test, or just a small typo?

File: src/__tests__/drop.test.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            expect(drop([1, 2, 3], NaN)).toEqual([1, 2, 3])
+        })
+    
+        test('should return the original array if `n` is null', () => {
+            expect(drop([1, 2, 3], null)).toEqual([1, 2, 3])
+        })
```
Comment (erictelkkala) [inline src/__tests__/drop.test.js:59]: I'd create one more test like this, but for "undefined" parameter

File: src/__tests__/isArrayLike.test.js
Comment (erictelkkala) [inline src/__tests__/isArrayLike.test.js:1]: I would specify the test names a bit more: e.g. "should return true for an empty string" and "should return true for a populated string"

File: src/__tests__/isArrayLike.test.js [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    describe('negative tests', () => {
+        describe('functions', () => {
+            test('should return false for functions', () => {
+                expect(isArrayLike(() => {})).toBe(false)
+            });
+            test('should return false for functions', () => {
+                expect(isArrayLike(function () {})).toBe(false)
+            });
+        })
+        describe('', () => {
```
Comment (erictelkkala) [inline src/__tests__/isArrayLike.test.js:52]: Missing the description

File: src/__tests__/isEmpty.test.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+            const typedArray = new Uint8Array([1, 2, 3])
+            expect(isEmpty(typedArray)).toBe(false)
+        })
+        test('should return false for arguments object with values', () => {
+            function testFunc() {
+              expect(isEmpty(arguments)).toBe(false)
+            }
+            testFunc(1, 2, 3)
+        })
+    })
```
Comment (erictelkkala) [inline src/__tests__/isEmpty.test.js:96]: Few tests here do not belong to another description function

File: src/__tests__/toNumber.test.js [hunk 6]
```diff
@@ -0,0 +1,138 @@
+import toNumber from '../toNumber';
+
+describe('toNumber', () => {
+    describe('positive tests', () => {
+      describe('numbers are numbers', () => {
+        test('13 should be 13', () => {
+          expect(toNumber(13)).toBe(13)
+        })
+        test('decimal 3.14 should be 3.14', () => {
```
Comment (erictelkkala) [inline src/__tests__/toNumber.test.js:9]:
```suggestion
        test('float 3.14 should be 3.14', () => {
```

File: src/__tests__/toNumber.test.js [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+        test('string is not number so it should return NaN', () => {
+          expect(toNumber('hello')).toBeNaN()
+        })
+        test('should return NaN for symbols', () => {
+          expect(toNumber(Symbol('test'))).toBeNaN()
+        })
+        test('return NaN for object that cannot be converted to number', () => {
+          expect(toNumber({})).toBeNaN()
+        })
+        test('return NaN for object that cannot be converted to number', () => {
```
Comment (erictelkkala) [inline src/__tests__/toNumber.test.js:78]:
```suggestion
        test('return NaN for an array that cannot be converted to number', () => {
```

File: src/__tests__/toString.test.js
Comment (erictelkkala) [inline src/__tests__/toString.test.js:1]: Some tests say "should be", while other say "turns into" and "convert"

**Review by erictelkkala — CHANGES_REQUESTED** [review]

File: src/__tests__/drop.test.js (hunk 1 above)
Comment (joniriikola) [reply src/__tests__/drop.test.js]: Fixed in commit [16760ea](https://github.com/erictelkkala/COMP.SE.200-2024-2025-1/pull/16/commits/16760ea4aa0f9b588a4f9a9a1a38b0c3ef560805)

File: src/__tests__/drop.test.js (hunk 3 above)
Comment (joniriikola) [reply src/__tests__/drop.test.js]: Fixed in commit [16760ea](https://github.com/erictelkkala/COMP.SE.200-2024-2025-1/pull/16/commits/16760ea4aa0f9b588a4f9a9a1a38b0c3ef560805)

File: src/__tests__/isArrayLike.test.js
Comment (joniriikola) [reply src/__tests__/isArrayLike.test.js:1]: Fixed in commit [16760ea](https://github.com/erictelkkala/COMP.SE.200-2024-2025-1/pull/16/commits/16760ea4aa0f9b588a4f9a9a1a38b0c3ef560805)

**Review by erictelkkala — APPROVED** [review]
Looks good to me now!

**Pull request merged**